velox/core/QueryConfig.h

```diff
@@ -36,6 +36,11 @@ class QueryConfig {
   static constexpr const char* kQueryMaxMemoryPerNode =
       "query_max_memory_per_node";
 
+  /// If true, Velox will use an ANSI-compliant dialect. For example, Velox will
+  /// throw a runtime exception instead of returning null results when the inputs
+  /// to a SQL operator or function are invalid.
+  static constexpr const char* kAnsiEnabled = "ansi_enabled";
```

Comment:
I assume this is specific to Spark functions. Presto functions are ANSI compliant already.

If so, let's add spark. prefix like other Spark-specific configs.

There is also a question about how to implement this without causing confusion or wrong results. It seems this change would allow user to request ANSI mode, but the results will not be compiant.

Reply:
@mbasmanova, thanks for your review. I will update the new config to indicate that it is intended for Spark use only.

There will be several upcoming PRs that rely on this configuration to adapt function implementations. The legacy mode and ANSI mode will coexist in Spark for a long period. We plan to introduce branching logic in relevant Velox functions, respecting this configuration to ensure ANSI compliance while minimizing code duplication. It's also possible to add separate ANSI-compliant functions if there isn't much shared code.

Reply:
This makes sense, but my question is about something else. How do we ensure that when ANSI is on the query returns correct results. If we land this change, then turning ANSI on will produce wrong results (same as with ANSI off). 

One option is to create an allow list of functions and special forms that support ANSI mode and fail the query if ANSI mode is ON, but query contains a function that's not in allow list. We would need to think about where to place such a check.

CC: @rui-mo

Reply:
The ANSI mode-controlled scenarios are described in Spark's documentation at https://github.com/apache/spark/blob/v4.0.0/docs/sql-ref-ansi-compliance.md, and they have a variety of effects on the detailed implementation. Given this, it might be difficult to cover every aspect by adding a check.

Since Gluten currently fallback for ANSI on, it would more of an experimental feature. We can also add a document in Velox to list the ANSI-compatible functions when supporting them, which should be rather limited for now.